<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>542</width>
    <height>416</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="dataFrame">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>211</width>
      <height>111</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(211, 211, 211);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>0</number>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>110</width>
       <height>91</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QPushButton" name="loadDataButton">
        <property name="text">
         <string>导入数据</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="value">
         <number>0</number>
        </property>
        <property name="textVisible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>1. 导入数据</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>20</y>
      <width>241</width>
      <height>111</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(211, 211, 211);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="text">
      <string>3. 选择模型</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="chooseModel">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>40</y>
       <width>121</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Decision Tree</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Random Forest</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>SVM</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>kNN</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>XGBoost</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>DNN</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="setParametersButton">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>70</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>设置参数</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>150</y>
      <width>211</width>
      <height>201</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(211, 211, 211);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>111</width>
       <height>32</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>200</height>
      </size>
     </property>
     <property name="text">
      <string>2. 数据预处理</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="indent">
      <number>-1</number>
     </property>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>181</width>
       <height>151</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="checkBox">
        <property name="text">
         <string>去除异常值</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkStandardScaler">
        <property name="text">
         <string>标准化</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QCheckBox" name="checkDimensionReduction">
          <property name="text">
           <string>数据降维</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="chooseDRMethod">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>PCA</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>NMF</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="startPreprocess">
        <property name="text">
         <string>预处理</string>
        </property>
        <property name="iconSize">
         <size>
          <width>16</width>
          <height>16</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>150</y>
      <width>241</width>
      <height>201</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(211, 211, 211);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="trainButton">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>训练</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="trainState">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>120</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>数据未加载</string>
     </property>
    </widget>
    <widget class="QPushButton" name="predictButton">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>60</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>预测</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="predictResult">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>170</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_predict">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>170</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>准确率</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>4. 模型训练</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>542</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>关于</string>
    </property>
    <addaction name="showAuthorAction"/>
    <addaction name="showLibrariesAction"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <action name="showAuthorAction">
   <property name="text">
    <string>作者</string>
   </property>
  </action>
  <action name="showLibrariesAction">
   <property name="text">
    <string>使用库</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>showAuthorAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showAuthor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showLibrariesAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showLibraries()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadDataButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setParametersButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setParameters()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>predictButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>predictData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>433</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>379</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trainButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>trainData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>243</y>
    </hint>
    <hint type="destinationlabel">
     <x>246</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startPreprocess</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>preprocessData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showAuthor()</slot>
  <slot>showLibraries()</slot>
  <slot>loadData()</slot>
  <slot>setParameters()</slot>
  <slot>predictData()</slot>
  <slot>trainData()</slot>
  <slot>preprocessData()</slot>
 </slots>
</ui>
